stackit/internal/services/serverbackup/schedule/resource.go

```diff
@@ -0,0 +1,539 @@
+		RetentionPeriod: types.Int64Value(*schedule.BackupProperties.RetentionPeriod),
+		VolumeIds:       ids,
+	}
+	return nil
+}
+
+// If already enabled, just continues
+func enableBackupsService(ctx context.Context, model *scheduleResourceModel, client *serverbackup.APIClient) error {
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+	enableServicePayload := serverbackup.EnableServicePayload{}
+
+	tflog.Debug(ctx, "Enabling server backup service")
+	err := client.EnableService(ctx, projectId, serverId).EnableServicePayload(enableServicePayload).Execute()
+	if err != nil {
+		if strings.Contains(err.Error(), "Tried to activate already active service") {
+			tflog.Debug(ctx, "Service for server backup already enabled")
+			return nil
+		}
+		return fmt.Errorf("Failed to enable server backup service: %w", err)
+	}
+	tflog.Info(ctx, "Enabled server backup service")
+	return nil
+}
+
+// Disables only if no backup schedules are present and no backups are present
+func disableBackupsService(ctx context.Context, model *scheduleResourceModel, client *serverbackup.APIClient) error {
+	tflog.Debug(ctx, "Disabling server backup service (in case there are no backups and no backup schedules)")
+
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+
+	tflog.Debug(ctx, "Checking for existing backup schedules")
+	schedules, err := client.ListBackupSchedules(ctx, projectId, serverId).Execute()
+	if err != nil {
+		return fmt.Errorf("Failed to disable server backup service: %w", err)
+	}
+	if *schedules.Items != nil && len(*schedules.Items) > 0 {
+		tflog.Debug(ctx, "Backup schedules found - will not disable server backup service")
+		return nil
+	}
+
+	tflog.Debug(ctx, "Checking for existing backups")
+	backups, err := client.ListBackups(ctx, projectId, serverId).Execute()
+	if err != nil {
+		return fmt.Errorf("Failed to disable server backup service: %w", err)
+	}
+	if *backups.Items != nil && len(*backups.Items) > 0 {
+		tflog.Debug(ctx, "Backups found - will not disable server backup service")
+		return nil
+	}
+
+	err = client.DisableService(ctx, projectId, serverId).Execute()
+	if err != nil {
+		return fmt.Errorf("Failed to disable server backup service: %w", err)
+	}
+	tflog.Info(ctx, "Disabled server backup service")
+	return nil
+}
+
+func toCreatePayload(ctx context.Context, model *scheduleResourceModel) (*serverbackup.CreateBackupSchedulePayload, error) {
+	if model == nil {
+		return nil, fmt.Errorf("nil model")
+	}
+
+	var backupProperties serverbackup.BackupProperties
+	if model.BackupProperties != nil {
+		ids := []string{}
+		if !(model.BackupProperties.VolumeIds.IsNull() || model.BackupProperties.VolumeIds.IsUnknown()) {
+			for _, elem := range model.BackupProperties.VolumeIds.Elements() {
+				val, err := conversion.ToString(ctx, elem)
+				if err != nil {
+					return nil, fmt.Errorf("Error by converting volume id: %w", err)
+				}
+				ids = append(ids, val)
+			}
```

Comment:
We have a util `ListValuetoStringSlice` that might be useful to simplify this logic